platform/AppStoreConnect/Sources/AppStoreConnect/Builds.swift

```diff
@@ -0,0 +1,105 @@
+import AppStoreConnect_Swift_SDK
+import Foundation
+
+enum BuildStatus {
+    case notYetUploaded
+    case processing
+    case invalid
+    case failed
+    case valid(Build)
+}
+
+extension AppStoreConnect {
+    
+    /// Waits until a given build is processed and is ready to be used.
+    /// - Parameters:
+    ///   - version: The version of the build to look for
+    ///   - buildNumber: The build number for the given version to look for
+    ///   - appID: The id of the app for which to search the build for
+    /// - Returns: The processed build if found, or throws error if something failed
+    func waitUntilBuildIsProcessed(forVersion version: String,
+                                   buildNumber: Int,
+                                   forApp appID: String) throws -> Build {
+        let maxNumberOfRequests = 10
+        let processingBuildCheckInterval: TimeInterval = 20
```

Comment:
So we're waiting up to 200 seconds? That's just about three minutes - if it's that short, I'm thinking we might as well do a single sleep (in the GA workflow probably, before invoking the make target), to reduce complexity. But wouldn't insist on it, just an idea for simplification.

Reply:
200 seconds are set based on one of my small test apps, not sure yet how long it would take for PolyPod.
An alternative is to wait longer after first check. Like, if the build is not yet uploaded or is still processing, to wait 2 minutes or so, and then make shorter periodical checks.

Reply:
Yeah, then nevermind my comment, let's keep the logic here. There's a couple of edge cases I can think of in which case the publishing would simply fail - e.g. when Apple needs to review our build first...

Reply:
Not quite, Apple first is processing the build. A build is unusable for any operations, until it is processed. Still, I am interested in other edge cases you might see :)